src/main/kotlin/org/onflow/flow/sdk/crypto/Crypto.kt

```diff
@@ -216,22 +216,26 @@ internal class SignerImpl(
 ) : Signer {
     override fun sign(bytes: ByteArray): ByteArray {
         val signature: ByteArray
+        val ecdsaSign: Signature
 
-        if (hashAlgo == HashAlgorithm.KECCAK256 || hashAlgo == HashAlgorithm.KMAC128) {
-            // Handle Keccak-256 and KMAC128 separately
-            val hash = hasher.hash(bytes)
-            val ecdsaSign = Signature.getInstance("NONEwithECDSA")
-            ecdsaSign.initSign(privateKey.key)
-            ecdsaSign.update(hash)
-            signature = ecdsaSign.sign()
-        } else {
-            // Handle other algorithms with valid IDs
-            val ecdsaSign = Signature.getInstance(hashAlgo.id)
-            ecdsaSign.initSign(privateKey.key)
-            ecdsaSign.update(bytes)
-            signature = ecdsaSign.sign()
+        val hash = hasher.hash(bytes)
+
+        // Determine the appropriate Signature instance based on the hash algorithm
```

Comment:
@tarakby Refactored `HashAlgorithm` to remove the id field and move logic to the signer

Reply:
@tarakby understood, I have disabled the 384 versions and updated the hashing algorithm codes as per https://cadence-lang.org/docs/language/crypto. I think a follow up PR is also needed here to remove the 384 versions from the HashingAlgorithms enum so that the index field is adjusted as well: https://github.com/onflow/flow/blob/master/openapi/access.yaml Thanks!

Reply:
Thank you for the changes 🙏🏼 
Regarding disabling the 384-bits versions, I only required them to be disabled from the ECDSA signing function, but I'm fine with them being enabled in the repo as a hash function, in case that makes your work easier or avoids the breaking change. You are more familiar with the repo than I am so I trust you choose the best option. From my side, I only wanted to discard them from being used in ECDSA.

Reply:
Understood, I've just gone through and updated the Cadence indices for the `index` field. For the sake of not impacting protobuf classes (which the Go SDK is also using) and tests running against the emulator II have left the 384-bit versions in the HashingAlgorithm enum. 

Looking back at our previous conversation about `code` and `index`, I think the reason the 256-bit and 384-bit versions were assigned the same code was to essentially disable the 384-bit versions (because they both point to the code of the corresponding 256-bit algorithm). For now I've assigned all of them a unique code, which in theory should not impact anyone using the SDK unless they are mistakenly using a 384-bit algorithm but protobuf is actually pointing to and using the 256-bit one. I suppose if we really want to avoid a breaking change we can keep the 384-bit versions using `code` `3` and `5`... although I'm not sure if that's really a breaking change or just masking users' calling a wrong/unsupported enum :) 

Let me know what you think re. the code values, otherwise I think the rest should be up to date now. Thanks!

<img width="494" alt="Screenshot 2024-07-24 at 21 03 42" src="https://github.com/user-attachments/assets/93c5a51f-8636-46e6-810a-334ff5735807">

Reply:
Sweet, thanks for the updates 🙏🏼 

> I have left the 384-bit versions in the HashingAlgorithm enum
Yes this is great 👌🏼 

Here is the summary of the values all SDKs should use:
- Access API codes, in this SDK they are called `code` https://developers.flow.com/build/basics/accounts#signature-and-hash-algorithms . They are called "Access API" because that's what we can use for signing transactions, the Flow network doesn't recognize other hashings.
- Cadence indices, in this SDK they are called `index` https://cadence-lang.org/docs/language/crypto#hashing. It's what smart contracts on Flow can support, that's why the list is larger than the Access API. 

Given these specs :
 - `code` is correct for the supported algos. However it may be confusing for devs that the non-supported algos are attributed a unique code. It may make sense to give the non-supported algos the code `-1`, which means they are officially non-supported by the access API.
 - `index` in this PR is now correct 👌🏼 
 
 I can commit those changes and let you know to review ⌛

Reply:
Thanks, yes I think -1 would be a good idea

Reply:
@tarakby I went ahead and committed the fix as I'm already here, thanks!

Reply:
I have left the KMAC and KECCAK `code` values as were initially set, not sure if you have any specific preference as they don't seem to be defined here: https://developers.flow.com/build/basics/accounts#signature-and-hash-algorithms

```
 KMAC128("KMAC128", 256, 5, 5),
 KECCAK256("KECCAK256", 256, 6, 6);

```

Reply:
`-1` should be applied to any non-supported algo by the Access API, so that includes KMAC and KECCAK too. I already pushed the change for those :)